crates/astria-sequencer/src/state_ext.rs

```diff
@@ -41,6 +41,22 @@ pub(crate) trait StateReadExt: StateRead {
         Time::parse_from_rfc3339(&String::from_utf8_lossy(&bytes))
             .context("failed to parse timestamp from raw timestamp bytes")
     }
+
+    #[instrument(skip(self))]
+    async fn get_storage_version_by_height(&self, height: u64) -> Result<u64> {
+        let key: Vec<u8> = format!("storage_version/{height}").into();
```

Comment:
Does it make sense to have a function for key construction to defend against accidental refactors/mismatches between this and `put_storage_version_by_height` below?

```rust
fn make_storage_version_by_height_key(height: u64) -> String {
    format!("storage_version/{height}")
}
```
Also you can avoid the casting with
```rust
self.nonconsensus_get_raw(format!("storage_version/{height}").as_bytes())
```